crud/src/main/java/com/redhat/lightblue/crud/NonPersistedPredefinedFieldTranslatorFromJson.java

```diff
@@ -0,0 +1,69 @@
+/*
+ Copyright 2015 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.crud;
+
+import com.redhat.lightblue.metadata.EntityMetadata;
+import com.redhat.lightblue.metadata.Field;
+import com.redhat.lightblue.metadata.FieldTreeNode;
+import com.redhat.lightblue.metadata.Fields;
+import com.redhat.lightblue.metadata.ObjectField;
+import com.redhat.lightblue.metadata.PredefinedFields;
+import com.redhat.lightblue.util.JsonNodeCursor;
+import com.redhat.lightblue.util.Path;
+
+/**
+ * When a datasource does not store fields automatically generated by Lightblue, this extension of
+ * {@link TranslatorFromJson} will ensure that the fields and values are stripped out.
+ *
+ * @author dcrissman
+ *
+ * @param <T> - The target entity type that the specific datastore knows how
+ * to interact with.
+ */
+public abstract class NonPersistedPredefinedFieldTranslatorFromJson<T> extends TranslatorFromJson<T>{
+
+    public NonPersistedPredefinedFieldTranslatorFromJson(EntityMetadata entityMetadata) {
+        super(entityMetadata);
+    }
+
+    @Override
+    protected void translate(JsonNodeCursor cursor, Object target) {
+        Path path = cursor.getCurrentPath();
+        FieldTreeNode fieldNode = entityMetadata.resolve(path);
```

Comment:
> It would make sense to fix it. We need to figure out if we have code that relies on that exception before trying to fix it.

I created #786 to address that issue. If it is acceptable to you, I'd like to move forward with this PR as is (pending moving to metadata namespace).